src/Components/ETRA/SignatureSignOff/index.tsx

```diff
@@ -0,0 +1,168 @@
+  signOffMode: boolean,
+  traId: number,
+  meetingId?: string,
+  meetingName: string,
+  attendees: IAttendees,
+  issues: Array<IIssue>,
+  signOff: ISignOff,
+  isSessionLive?:boolean
+}
+
+export interface ISaveMeetingState {
+  isAttemptingToSave: boolean;
+  isValid: boolean;
+  redirectToLandingPage: boolean;
+ 
+}
+
+export class SignatureSignOff extends React.Component<ISaveMeetingProps, ISaveMeetingState> {
+  public static contextType = ServiceContext;
+  private readonly signoffMeeting: ISignOffMeetingUseCase;
+  private readonly createMeeting: ICreateMeetingUseCase;
+
+  public constructor(props: ISaveMeetingProps, context: IServiceProvider) {
+    super(props, context);
+    this.signoffMeeting = context.get<ISignOffMeetingUseCase>("ISignOffMeetingUseCase");
+    this.createMeeting = context.get<ICreateMeetingUseCase>("ICreateMeetingUseCase");
+    
+    this.state = {
+      isAttemptingToSave: false,
+      isValid: this.checkIsValid(this.props),
+      redirectToLandingPage: false
+    }
+  }
+ 
+  componentWillReceiveProps(newProps: ISaveMeetingProps){
+    this.setState({isValid: this.checkIsValid(newProps)})
+   
+  }
+
+  private checkIsValid(props: ISaveMeetingProps){
+    if(isNaN(props.attendees.attendees) || props.attendees.attendees < 1){
+      return false;
+    }
+    
+    return true;
+  }
+
+  getMeetingModel = () : IMeetingModel => {
+    return new MeetingModel(
+      this.props.traId,
+      this.props.meetingName,
+      this.props.issues, 
+      this.props.attendees, 
+      this.props.signOff,
+      this.props.meetingId, 
+    );
+  }
+
+  handleSaveDraft = () => {
+    this.setState({ isAttemptingToSave: true });
+    /*const successful = this.saveMeetingDraft.Execute(this.getMeetingModel());
+
+    if (successful) {
+      this.setState({ redirectToLandingPage: true });
+    }
+    else {
+      this.setState({ isAttemptingToSave: false });
+    }*/
+  }
+
+  getMeetingModelWithSignatureTrimmed = (): IMeetingModel => {
+    const trimmedSignatureSignoff:ISignOff = {
+      signature:this.props.signOff.signature.slice(22) ,
+      name: this.props.signOff.name,
+      role: this.props.signOff.role
+    }
+    return new MeetingModel(
+      this.props.traId,
+      this.props.meetingName,
+      this.props.issues, 
+      this.props.attendees, 
+      trimmedSignatureSignoff,
+      this.props.meetingId, 
+    )
+  }
+
+  getMeetingSignOffModel = () : IMeetingSignOffModel => {
+    return new MeetingSignOffModel(
+      this.props.meetingId!,
+      this.props.signOff
+    );
+  }
+
+  handleSaveMeeting = async () => {
+    let successful;
+
+    this.setState({ isAttemptingToSave: true });
+
+    if(this.props.signOffMode){
+      successful = await this.signoffMeeting.Execute(this.getMeetingSignOffModel());
+    }else{
+      successful = await this.createMeeting.Execute(this.getMeetingModelWithSignatureTrimmed());
+    }
+
+    if (successful) {
+      //this.props.onReviewNow();
+    }
+
```

Comment:
should this have something in the if statement?

Reply:
See previous comment. I will clean these up when the application is complete